src/ai/backend/manager/services/model_serving/services/auto_scaling.py

```diff
@@ -179,3 +183,28 @@ async def delete_endpoint_auto_scaling_rule(
         return DeleteEndpointAutoScalingRuleActionResult(
             success=True,
         )
+
+    async def search_auto_scaling_rules(
+        self, action: SearchAutoScalingRulesAction
+    ) -> SearchAutoScalingRulesActionResult:
+        """Searches endpoint auto scaling rules."""
+        await self.check_requester_access(action.requester_ctx)
+
+        if action.requester_ctx.user_role == UserRole.SUPERADMIN:
+            result = await self._admin_repository.search_auto_scaling_rules_force(
+                querier=action.querier,
+            )
+        else:
+            result = await self._repository.search_auto_scaling_rules_validated(
+                querier=action.querier,
+                user_id=action.requester_ctx.user_id,
+                user_role=action.requester_ctx.user_role,
+                domain_name=action.requester_ctx.domain_name,
+            )
```

Comment:
Can we keep only one repository? Splitting this into two repositories was my mistake.